### DISCUSSION TITLE
Installing packages in RStudio

### DISCUSSION THREAD

**Discussion by danhps** [neurodesk/neurodesk.github.io-hugo#384] [New Tool]
Is there some way that I can install a package in RStudio in a way that it will still be installed the next time I open up NeuroDesk? If not, is there any chance the package RMINC (https://github.com/Mouse-Imaging-Centre/RMINC/tree/master) could be installed? Thanks!

Dan

**Comment by stebo85** [comment]
Dear @danhps, 

yes, both things are possible :) how are you accessing neurodesk and which version are you using?

**Comment by danhps** [comment]
Hi @stebo85! Thanks for your reply. Please bear with me because I am very new to this. I open the NeuroDesk App from my (Windows) desktop, select "Open Local Neurodesk", select "Neurodesktop" from the Launcher, select "Desktop Dynamic Resolution", select "rstudio 1.4.1106" from the Start menu, which opens up rstudio in Terminal, and finally type in "rstudio" in the Terminal to launch the program. 

Ideally I would like to be able to save my session so I could come back to it again and it would be the same as I left it, but if that's not possible then I would really appreciate being able to RMINC (and MRIcrotome) so that they remain installed somehow.

Is that the info you needed? If not please let me know and I'll get it!

Thanks again!

Dan

**Reply by danhps** [reply]
Update: I just went back into rstudio as I described above, and RMINC is already installed. Yesterday, I had to reinstall it. Is there something you did on the backend, or did I do something different that I didn't notice?

**Reply by stebo85** [reply]
Dear @danhps,

cool - that's great to hear it's working. The persistent home directory is a new feature we added a few weeks ago, so it's possible that you updated to the latest version of neurodeskapp (1.2.1) which made it work.

thank you 
Steffen

**Comment by danhps** [comment]
Thanks! Really appreciate the response.

**Comment by danhps** [comment]
@stebo85, some processes I was to do using RMINC are too much for my little laptop so I log into a remote service (I use the one in Sydney because I'm in Melbourne and I imagine it'll be faster, not sure if that's accurate). Every time I do this, I have to re-upload all my files and re-install RMINC. Is there a way that RMINC could be included in the packages that are installed on these remote desktops, and is there some way for me to have my files remain on the remote desktops? I imagine maybe not, since they wouldn't want junk accumulating on their servers...

**Reply by stebo85** [reply]
Dear @danhps,

The best option for you would be to use the Nectar virtual desktop service: https://www.neurodesk.org/docs/getting-started/neurodesktop/nectar/

 It persists your data and provides you with a lot of resources :)

Kind regards
Steffen

**Reply by stebo85** [reply]
and yes, I will include RMINC in the next release into the rstudio container

**Comment by stebo85** [comment]
Dear @danhps 

RMINC is now added to the latest Rstudio 2023.09.1 update in Neurodesk:

<img width="2949" alt="image" src="https://github.com/NeuroDesk/neurodesk.github.io/assets/4021595/3ea5db91-3ef1-4fed-bcf1-0ada352b052f">

To get the updated entry in the application menu you have to click the update button in the main menu.

Can you test and and let me know if this works for you? 

Also, did nectar virtual desktop service work for you and solve your problems?

**Reply by danhps** [reply]
Hi @stebo85, sorry, I've been away, but now I'm back and I really appreciate your help! I'm going to check the update and Nectar and let you know how it goes. Thanks and sorry again!

Dan

**Reply by stebo85** [reply]
Dear @danhps,

No worries :) Let us know how you go and if you need anything else!

Cheers
Steffen

**Reply by danhps** [reply]
Hi @stebo85, just wanted to report that I've been using RMINC on Neurodesk through Nectar for over a week now and it's working great! Thanks a lot for your help.

The one thing I would still like to request, and it is very minor, is if you could consider adding the R package MRIcrotome to your list of included R packages for your next upgrade. I currently run devtools::install_github("Mouse-Imaging-Centre/MRIcrotome") to install it, and it doesn't take much time at all, so it's a very minor suggestion, but just something to add to your list if you like.

**Reply by stebo85** [reply]
Dear @danhps - great to hear it's working well for you :)

You can add this package to our containers yourself very easily. Could you please propose the change to this file? https://github.com/NeuroDesk/neurocontainers/blob/master/recipes/rstudio/dependencies.R

I am asking you to propose the change because to keep the project sustainably funded, it would help us a lot to show that users actively contribute to the project and that it's not just the core team of developers making changes to the code! The more contributors the better :)

Thank you
Steffen

**Reply by stebo85** [reply]
Dear @danhps - just wondering, if you have a chance of proposing this update on github? It would just be clicking on the little edit button in https://github.com/NeuroDesk/neurocontainers/blob/master/recipes/rstudio/dependencies.R, entering a new line:
```
if("MRIcrotome" %in% rownames(installed.packages()) == FALSE) {devtools::install_github("Mouse-Imaging-Centre/MRIcrotome", upgrade_dependencies = FALSE)}
```
and proposing the change through a pull request. This would then also give you credit as a contributor :)

**Comment by danhps** [comment]
Dear @stebo85, thanks for the reminder! I just did it. Let me know if it worked, I'm a very inexperienced GitHub user.

**Reply by stebo85** [reply]
that worked :) It's rebuilding now and once the new container is ready I let you know so you can test it!

**Reply by stebo85** [reply]
Dear @danhps - can you execute this in your neurodesktop and let me know if the newly added packages works as expected?
```
bash /neurocommand/local/fetch_and_run.sh rstudio 2023.12.1 20240209
```

**Reply by stebo85** [reply]
@danhps - where you able to test it and is it working as expected?
